hooks/gate-scripts/lib/commit_scope.py

```
@@ -0,0 +1,373 @@
+    `.claude` on any caller that legitimately overrides it.
+    """
+    dirs = {os.path.join(repo, state_dir), os.path.join(repo, ".claude")}
+    for d in dirs:
+        for name in ("settings.json", "settings.local.json"):
+            path = os.path.join(d, name)
+            if not os.path.exists(path):
+                continue
+            try:
+                with open(path, "rb") as fh:
+                    data = json.loads(fh.read().decode("utf-8", "surrogateescape"))
+            except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)